electron-src/process/js/parser.js

```diff
@@ -26,24 +21,22 @@ class Parser{
             let pref = t.tutorPreferences;
 
             let courses = pref.coursePreferences;
-            if (courses.PREFER.length !== 0) {
-                tempTutor.coursePreference = courses.PREFER.map((c)=>c.name);
-            }else{
-                tempTutor.coursePreference = "";
-            }
+
+            tempTutor.coursePreference = _.map(courses.PREFER, (c)=>c.name);
+            tempTutor.courseWilling = _.map(courses.WILLING, (c)=>c.name);
 
             let days = pref.shiftPreferences;
-            if (days.PREFER.length !== 0) {
-                tempTutor.shiftPreference = days.PREFER[0].day;
-            }
+
+
+            tempTutor.shiftPreference = _.map(days.PREFER, (d)=>d.day);
+            tempTutor.shiftWilling = _.map(days.WILLING, (d)=>d.day);
+
             let fre = pref.shiftFrequencyPreferences;
-            if (fre.PREFER.length !== 0){
-                tempTutor.shiftFrequency = fre.PREFER;
-            }
-            else{
-                tempTutor.shiftPreference = "";
-            }
 
+            tempTutor.shiftAmountPreferred = fre.PREFER;
+            tempTutor.shiftAmountWilling = fre.WILLING;
+
+            tempTutor.notInclude = false;
```

Comment:
Exclude? To mimic the frontend verbiage? Not vital or a required change, but just a suggestion for consistency.